rfc-149-switch-to-per-page-asset-loading.md

```diff
@@ -0,0 +1,208 @@
+
+### Shared assets
+
+The assets should be placed in a folder that allows each application to reference them. The current behaviour has each application place their assets in a different folder - `/assets/collections`, `/assets/finder-frontend` etc. This prevents the browser cache from being used effectively as `/assets/collection/accordion.css` will be downloaded even if `/assets/finder-frontend/accordion.css` has already been downloaded.
+
+A version identifier should be added to the component asset filename to help identify which version of the components gem is being used - for example `gem-v29-2-1-accordion.css`. Similarly, the bespoke app assets should be renamed to their application - so collections would have `collections.css` and `collections.js` - to make it easier to see which stylesheet comes from which application.
+
+The assets would ideally be placed in a shared folder as per [RFC #91][sharing_assets], so would be able to be used across GOV.UK irrespective of the rendering application. Since this hasn't been implemented yet, there would need to be some work for this to happen - and likely this would be delayed to coincide with the replatforming work.
+
+This work should not block the serving of individual assets as there are advantages that don't rely on having a shared folder.
+
+## Considerations
+
+### Reduced compressibility when serving individual files
+
+GOV.UK uses [Brotli compression][enable_brotli] when serving assets - this is a compression method that works really well on repetitive text like HTML, CSS, and JavaScript. Serving individual files will reduce the amount of compression available because the files will have less repetition in them. Since each page will only be serving the component assets required on the page - rather than the components used in the entire application - there should be a reduction in asset size.
+
+To determine whether this is correct, I looked at the difference in CSS size for the top 100 most visited pages on GOV.UK over the past year (1 July 2021 to 31 June 2022). There was reduction in CSS size for pages[^1].
+
+The largest change in component CSS size is -12.2kB, the smallest is -1kB, and the average decrease is 5.8kB; the graph shows the top 100 pages and the reduction in the amount of CSS for each page:
+
+![](rfc-149/css-size-for-top-100-most-visited-pages-on-govuk.svg)
+
+
+This means that all visitors will see a smaller page size, regardless of whether they visit one page or multiple pages. Six out of ten tracked visitors to GOVUK only view a single page.
+
+Concatenating the assets can provide a benefit for those that visit multiple pages - but this doesn't work on GOVUK due to the number of rendering apps used and the current lack of a shared space to put assets.
```

Comment:
```suggestion
This means that all visitors will see a smaller page size, regardless of whether they visit one page or multiple pages. Six out of ten tracked visitors to GOV.UK only view a single page.

Concatenating the assets can provide a benefit for those that visit multiple pages - but this doesn't work on GOV.UK due to the number of rendering apps used and the current lack of a shared space to put assets.
```

Reply:
Again, good spot! Fixed in https://github.com/alphagov/govuk-rfcs/pull/152/commits/35f516e8f71c2a0dc69e67367f0ac35056a6d722